imgcreate/live.py

```diff
@@ -1027,15 +1027,213 @@ def _get_excluded_packages(self):
         return ["kernel.ppc"] + \
                ppcLiveImageCreator._get_excluded_packages(self)
 
+class aarch64LiveImageCreator(LiveImageCreatorBase):
+    """ImageCreator for arm machines"""
+    def __init__(self, *args, **kwargs):
+        LiveImageCreatorBase.__init__(self, *args, **kwargs)
+
+    def _get_xorrisofs_options(self, isodir):
+    # aarch64 system not have legasy boot, only UEFI
+    # dir isolinux not needed, put efiboot.img to /EFI/BOOT dir on iso
+        options = [ "-no-emul-boot", "-graft-points",
+                    "-allow-lowercase"]
+        options.extend([ "-eltorito-alt-boot",
+                         "-e", "EFI/BOOT/efiboot.img",
+                         "-no-emul-boot",
+                         "-J", "-D", "-R"])
+        return options
+
+    def _get_required_packages(self):
+        return ["grub2-efi"] \
+               + LiveImageCreatorBase._get_required_packages(self)
+
+    def __copy_kernel_and_initramfs(self, isodir, version, index):
+    # Arm system not have legasy boot, only UEFI
+    # dir isolinux not needed, put vmlinuz and initrd to /LiveOS dir on iso
+        bootdir = self._instroot + "/boot"
+        makedirs(isodir+"/LiveOS/")
+        shutil.copyfile(bootdir + "/vmlinuz-" + version,
+                        isodir + "/LiveOS/vmlinuz" + index)
+        shutil.copyfile(bootdir + "/initrd-" + version + ".img",
+                        isodir + "/LiveOS/initrd" + index + ".img")
+
+    def __is_default_kernel(self, kernel, kernels):
+        if len(kernels) == 1:
+            return True
+
+        if kernel == self._default_kernel:
+            return True
+
+        if kernel.startswith(b"kernel-") and kernel[7:] == self._default_kernel:
+            return True
+
+        return False
+
+    def __copy_efi_files(self, isodir):
+        """ Copy the efi files into /EFI/BOOT/
+            If any of them are missing, return False.
+            requires:
+              gcdaa64.efi or grubcd.efi
+              fonts/unicode.pf2
+        """
+        fail = False
+        # BOOTAA64.EFI - not signed, we use grub2 directly
+        # TODO: use shim as primary bootloader when secureboot needed
+        # temporary create dir with BOOTAA64.EFI for efiboot.img
```

Comment:
Signed bootloaders exist in Fedora and CentOS/RHEL for AArch64, please adapt this accordingly.

The packages with signed bootloader content are:

* `shim-aa64`
* `grub2-efi-aa64`